src/test/e2e/29_mismatched_architectures_test.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestMismatchedArchitectures ensures that zarf produces an error
+// when the package architecture doesn't match the target cluster architecture.
+func TestMismatchedArchitectures(t *testing.T) {
+	t.Log("E2E: Mismatched architectures")
+	e2e.SetupWithCluster(t)
+	defer e2e.Teardown(t)
+
+	var (
+		mismatchedArch         = e2e.SetMismatchedArch()
```

Comment:
```suggestion
		mismatchedArch         = e2e.GetMismatchedArch()
```
If the above suggestion is taken this sill be needed.